src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -525,3 +525,31 @@ func AddImagePullSecretToDefaultServiceAccount(ctx context.Context, client klien
 	}
 	return nil
 }
+
+func VerifyCloudAPIAdaptorLogs(ctx context.Context, client klient.Client, t *testing.T, pod *v1.Pod, expectedSuccessMessage string) error {
+	var podlist v1.PodList
+	var podLogString string
+
+	if err := client.Resources("confidential-containers-system").List(ctx, &podlist); err != nil {
+		return err
+	}
+	for _, pod := range podlist.Items {
+		if pod.Labels["app"] == "cloud-api-adaptor" {
+			podLogString, _ = GetPodLog(ctx, client, pod)
+			break
+		}
+	}
+	if strings.Contains(podLogString, expectedSuccessMessage) {
+		t.Logf("cloud-api-adaptor logs are matching with the expected message")
+	} else {
+		yamlData, err := yaml.Marshal(pod.Status)
+		if err != nil {
+			t.Fatalf("Error marshaling pod.Status to JSON: %s", err.Error())
+		} else {
+			t.Logf("Current Pod State: %v", string(yamlData))
+			t.Fatal("failed due to unknown reason")
+		}
+		return errors.New("failed due to unknown reason")
```

Comment:
Apart from the pod app being hard-coded (which might be enough of an issue to make this not re-usable), this looks pretty similar to `ComparePodLogString`, so I wonder if you could call `ComparePodLogString` from within this to reduce duplication?